homeassistant/components/weatherflow/sensor.py

```diff
@@ -348,6 +369,22 @@ def __init__(
                 description.imperial_suggested_unit
             )
 
+    @property
+    def icon(self) -> str | None:
+        """Get icon."""
+
+        value = (
+            int(self.native_value)
+            if self.native_value is not None
+            and self.native_value != "none"
+            and isinstance(self.native_value, (int, float, str))
+            else None
+        )
+
+        if value and self.entity_description.icon_fn is not None:
+            return self.entity_description.icon_fn(value)
+        return None
```

Comment:
Should we return super().icon instead?

Reply:
Based on other PRs I think we return `None` ... but it's a Good question. In testing at least - setting the value to `None` uses the default icon for the sensor type.

Reply:
I couldn't remember on this either, but if you're seeing the correct icons still, I'll assume it works